universal-login-example/src/components/RecoverAccount.js

```diff
@@ -0,0 +1,61 @@
+import RecoverAccountView from '../views/RecoverAccountView';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {Wallet} from 'ethers';
+import DEFAULT_PAYMENT_OPTIONS from '../../config/defaultPaymentOptions';
+
+class RecoverAccount extends Component {
+  constructor(props) {
+    super(props);
+    this.sdk = this.props.services.sdk;
+    this.identityService = this.props.services.identityService;
+    this.emitter = this.props.services.emitter;
+    this.state = {
+      backupCode: '',
+      isLoading: false
+    };
+  }
+
+  async onCancelClick() {
+    const {emitter} = this.props.services;
+    emitter.emit('setView', 'Login');
+    this.identityService.cancelSubscription();
+
+    const {identityService} = this.props.services;
+    const identityAddress = identityService.identity.address;
+    const {address} = new Wallet(identityService.privateKey);
+    const {sdk} = identityService;
+    await sdk.denyRequest(identityAddress, address);
+  }
+
+  async onRecoverClick() {
+    this.setState({isLoading: true});
+    const {identityService, sdk} = this.props.services;
+    let wallet = await Wallet.fromBrainWallet(this.identityService.identity.name, this.state.backupCode);
+    await sdk.addKey(identityService.identity.address, identityService.deviceAddress, wallet.privateKey, DEFAULT_PAYMENT_OPTIONS);
```

Comment:
There's no error handling here. When you input an incorrect recovery code, the app just keeps spinning forever

Reply:
Reviewed ✅